library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -1022,4 +1023,55 @@ class GroupTest {
         // Next syncAllGroups will not include the inactive group
         assertEquals(numGroups, 2u)
     }
+
+    @Test
+    fun testGroupDisappearingMessages() {
+        // Can enable message expiration on create
+        val boGroup = runBlocking {
+            boClient.conversations.newGroup(
+                listOf(alix.walletAddress),
+                messageDisappearingSettings = MessageDisappearingSettings(0L, 10L),
+            )
+        }
+        val messageId = runBlocking {
+            boGroup.send("howdy")
+        }
+        runBlocking { alixClient.conversations.syncAllConversations() }
+        var message = boClient.findMessage(messageId)
+        var alixMessage = alixClient.findMessage(messageId)
+        val alixGroup = alixClient.findGroup(boGroup.id)
+        assert(message != null)
+        assert(alixMessage != null)
+        assertEquals(runBlocking { boGroup.messages() }.size, 2)
+        assertEquals(runBlocking { alixGroup!!.messages() }.size, 1)
+        Thread.sleep(2000)
+        runBlocking {
+            boClient.conversations.syncAllConversations()
+            alixClient.conversations.syncAllConversations()
+        }
+        message = boClient.findMessage(messageId)
+        alixMessage = alixClient.findMessage(messageId)
+        assert(message == null)
+        assertEquals(runBlocking { boGroup.messages() }.size, 0)
+        assert(alixMessage == null)
+        assertEquals(runBlocking { alixGroup!!.messages() }.size, 0)
+
+        // Can disable message expiration
+        val messageDisappearingSettings = MessageDisappearingSettings(0L, 0L)
+        runBlocking {
+            alixGroup!!.updateMessageExpiration(messageDisappearingSettings)
```

Comment:
Is this how you turn off message disappearing settings? Seems like the update method in libxmtp should be nullable instead?

Reply:
We can't set the values to null, because if the value set to null, we can't detect it in the metaDataUpdate process.
but for removing the settings, you can call `remove_conversation_message_disappearing_settings`. that would set the default values for you.